feature 20 added scripts for 4 tutorial setup videos

georgemccabe opened the pull request:
This PR contains the scripts for the 4 METplus 4.0 Online Tutorial Setup videos, including supplemental files (PowerPoint slides) that were used in the video. I provided a PDF of one of them to preserve the formatting. The PowerPoint files are large, but we do want to be able to edit them. I also added template slides and template text to include in all future online tutorial videos.

## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Checked the web content

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review the content added to the repository and the videos that are found in the METplus Team Drive:
https://drive.google.com/drive/folders/1lTDnGHQBH8RXP0GC6ZeurIC2HHm-e51F

- [X] Please complete this pull request review by **10/20/2021**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

JohnHalleyGotway reviewed (changes requested):
@georgemccabe, I really like it that the feature branch is available for review on RTD:
https://metplus-training.readthedocs.io/en/feature_20_tutorial_setup/modules/Tutorial/index.html

Here's my feedback:
- Seeing it on that page, I do think we should break it out onto 4 separate pages - one for each video. Putting them one one page is just too much scrolling and having one page = one video will provide a simple, consistent user experience. I do really like it that you have it broken up into 4 separate sections on the left side navigation.

- Now that Jenny's uploaded them to YouTube, please drop in the actual links.

- It sure doesn't seem like we should save both a .pdf and .pptx version of the same slides in the repo. But I do see the nice formatting in the .pdf while the .pptx looks bad on my Mac. I sure wish we didn't have platform dependent behavior like this! Perhaps Google slides wouldn't be platform dependent? Any suggestions for how we should handle this?

- I have NOT actually gone through the tutorial instructions to make sure they work and am headed out for the weekend. But I can do so on Tuesday.

Thanks for working on this!

georgemccabe commented:
> 
> 
> @georgemccabe, I really like it that the feature branch is available for review on RTD: [metplus-training.readthedocs.io/en/feature_20_tutorial_setup/modules/Tutorial/tutorial_setup/tutorial_setup.html](https://metplus-training.readthedocs.io/en/feature_20_tutorial_setup/modules/Tutorial/tutorial_setup/tutorial_setup.html)

**I agree that it is really convenient, except when trying to reload changes and it uses a cached version. It is a minor inconvenience that comes with a great advantage!**

> 
> Here's my feedback:
> 
>     * Seeing it on that page, I do think we should break it out onto 4 separate pages - one for each video. Putting them one one page is just too much scrolling and having one page = one video will provide a simple, consistent user experience. I do really like it that you have it broken up into 4 separate sections on the left side navigation.

**I split up the content into 4 pages and updated the table of contents. Take a look at the feature branch on RTD and let me know what you think. Sphinx didn't like having nested table of contents, so I put the table of contents on the "Online Tutorial Topics" page with each video linked under a "METplus Tutorial Setup" sub-heading. It still structures the side menu nicely. Clicking on "METplus Tutorial Setup" in the menu just expands the menu instead of redirecting to a title page. I think that works well.**

> 
>     * Now that Jenny's uploaded them to YouTube, please drop in the actual links.

**I just updated the links for the videos.**

> 
>     * It sure doesn't seem like we should save both a .pdf and .pptx version of the same slides in the repo. But I do see the nice formatting in the .pdf while the .pptx looks bad on my Mac. I sure wish we didn't have platform dependent behavior like this! Perhaps Google slides wouldn't be platform dependent? Any suggestions for how we should handle this?

**I was torn here which is why I just provided both.  If updates aren't needed, the smaller PDF files are sufficient. We could upload the PDFs to the repo and upload the PPTX files to the shared drive and provide a link to the top directory that contains it? This would all the video to be reproducible as long as no updates are needed but still possible to retrieve the slides to update them as needed. What do you think?**

> 
>     * I have NOT actually gone through the tutorial instructions to make sure they work and am headed out for the weekend. But I can do so on Tuesday.

**I ran the instructions in the videos so they SHOULD work but it doesn't hurt for someone else to go through them to make sure.**

> 
> 
> Thanks for working on this!

You are welcome!

georgemccabe commented:
@JohnHalleyGotway, I also noticed that there was a warning in the documentation that wasn't being caught by the GHA run, so I made some updates to properly fail when a warning/error occurs, verified that the correct failure happened, then made a fix to remove the warning.

JohnHalleyGotway commented:
@georgemccabe since PDF's are static and not really editable, I don't see the advantage of including them in version control. There's always going to something to update in the future, so I'd rather store the editable pptx version instead of the static PDF's. But I do think it's really unfortunate that the pptx files do not render the same on Mac and Windows. And I'm wondering if Google Slides **would** render the same on both. If so, we may want to move to that for future videos.

I'll step through the videos myself this afternoon and complete the review today.

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_1.rst:
```diff
@@ -0,0 +1,137 @@
+The content in this video corresponds to this online tutorial section: https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/metplus-setup
+
+Next Video:
+If using a pre-configured environment such as Hera or Cheyenne: ***ADD URL FOR Session 1 Video 2a***
```
Since the videos are posted already, I think you can drop in these links now.

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_2a.rst:
```diff
@@ -0,0 +1,91 @@
+The content in this video corresponds to this online tutorial section: https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/metplus-setup/metplus-initial-setup
+
+Next Video:
+***ADD URL FOR Session 1 Video 3***
```
Add links.

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_2a.rst:
```diff
@@ -0,0 +1,91 @@
+
+(show slide 5 - Topics Covered in This Video)
+
+***Add clickable link to “Set Up User Workstation Environment” video***
```
Add link

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_2a.rst:
```diff
@@ -0,0 +1,91 @@
+
+(show slide 6 - links)
+
+***add clickable links for next video and useful links in description***
```
Add link

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_2b.rst:
```diff
@@ -0,0 +1,114 @@
+The content in this video corresponds to this online tutorial section: https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/metplus-setup/metplus-initial-setup
+
+Next Video:
+***ADD URL FOR Session 1 Video 3***
```
Add links here and below.

JohnHalleyGotway commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_3.rst:
```diff
@@ -0,0 +1,76 @@
+https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/metplus-setup/verify-environment-set-correctly
+
+Next Video:
+***ADD URL FOR ?***
```
Add links here and below.

georgemccabe commented on docs/modules/Tutorial/tutorial_setup/tutorial_setup_2a.rst:
```diff
@@ -0,0 +1,91 @@
+
+Developed for **METplus Version 4.0**.
+
+Preparation:
```
```suggestion
Preparation
```
